### PR TITLE
MATLAB: fixes in `make_consistent`, getting started examples using new interface

### PR DESCRIPTION
**Pull request by sandmaennchen**
This PR fixes some issues in `make_consistent`
* `has_x0` has not been set correctly
* allow for empty CasADi symbolic in model

.. and ports the examples in `getting_started` to the new interface. The old extensive example has been moved to `legacy_interface` to keep it for testing.

### CODE REVIEW

File: interfaces/acados_matlab_octave/AcadosOcp.m [hunk 1]
```diff
@@ -257,6 +257,13 @@ function make_consistent(self)
                 dims.nbxe_0 = 0;
             end
 
+            % check if has_x0 is true
+            % if true ineqaulities are set to equalities
+            if dims.nx == dims.nbx_0 && all(sort(reshape(constraints.idxbx_0, 1, [])) == 0:(dims.nx-1)) && all(constraints.lbx_0 == constraints.ubx_0)
+                constraints.has_x0 = true;
+                dims.nbxe_0 = dims.nx;
```
Comment (FreyJo) [inline interfaces/acados_matlab_octave/AcadosOcp.m:264]: If we add this, it is not possible to have an x0 constraint without marking it as an equality constraint. Right?
This concerns me a bit, since I think this is sometimes desired.
Why did you add this?
Could we just have the check in the transription from the legacy interface maybe?

File: examples/acados_matlab_octave/getting_started/minimal_example_ocp.m [hunk 2]
```diff
@@ -15,7 +15,7 @@
 % this list of conditions and the following disclaimer in the documentation
 % and/or other materials provided with the distribution.
 %
-% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
```
Comment (FreyJo) [inline examples/acados_matlab_octave/getting_started/minimal_example_ocp.m:18]: probably unintended change.

File: examples/acados_matlab_octave/getting_started/minimal_example_closed_loop.m [hunk 3]
```diff
@@ -28,146 +28,130 @@
...
-model_path = fullfile(pwd,'..','pendulum_on_cart_model');
-addpath(model_path)
-
 check_acados_requirements()
 
 % initial state
 x0 = [0; 0; 0; 0];  % start at stable position
 
+%% SOLVER OPTIONS
+solver_options = AcadosOcpOptions();
```
Comment (FreyJo) [inline examples/acados_matlab_octave/getting_started/minimal_example_closed_loop.m:39]: In the Python examples, we typically do create `ocp = AcadosOcp()`  and then set options:
`ocp.solver_options.* = ...`
I somehow prefer this a bit.
Also maybe we can check if you can overwrite `ocp.solver_options` by something that is not an `AcadosOcpOptions` object.

**Review by FreyJo — APPROVED** [review]
Added minor comments.
Great to have getting started examples fully migrated :tada:

File: examples/acados_matlab_octave/getting_started/get_pendulum_on_cart_AcadosModel.m [hunk 4]
```diff
@@ -36,7 +36,14 @@
     nu = 1;
 
     %% system parameters
-    M = 1;    % mass of the cart [kg]
+    if nargin > 1 && varargin{2} % parametric model
```
[the remark this answers is not in the document]
Comment (FreyJo) [reply examples/acados_matlab_octave/getting_started/get_pendulum_on_cart_AcadosModel.m:43]: This has already been fixed. I started the review too early.

File: examples/acados_matlab_octave/getting_started/minimal_example_closed_loop.m (hunk 3 above)
Comment (sandmaennchen) [reply examples/acados_matlab_octave/getting_started/minimal_example_closed_loop.m:39]: but somehow we don't do it for the model :shrug:

Comment (FreyJo) [reply examples/acados_matlab_octave/getting_started/minimal_example_closed_loop.m]: Right, it is somehow more common to have a function that sets up the model. But I think it is nice that the example matches the python one closely now.

File: interfaces/acados_matlab_octave/AcadosModel.m [hunk 5]
```diff
@@ -190,45 +190,47 @@ function make_consistent(obj, dims)
...
+            elseif iscolumn(obj.p) || (isa(obj.p, 'casadi.SX') == isSX && length(obj.p) == 0)
                 dims.np = size(obj.p, 1);
             else
                 error('model.p should be column vector.');
             end
 
             if isempty(obj.xdot)
                 obj.xdot = empty_var;
-            elseif ~iscolumn(obj.xdot) || size(obj.xdot, 1) ~= dims.nx
+            elseif ~(isa(obj.z, 'casadi.SX') == isSX && length(obj.z) == 0) && (~iscolumn(obj.xdot) || size(obj.xdot, 1) ~= dims.nx)
```
Comment (FreyJo) [inline interfaces/acados_matlab_octave/AcadosModel.m:218]: Is this a typo? I think z should be xdot.

File: interfaces/acados_matlab_octave/setup_AcadosOcp_from_legacy_ocp_description.m [hunk 6]
```diff
@@ -377,6 +377,14 @@
     else
         error(['integrator ', opts_struct.sim_method, ' not support for templating backend.'])
     end
+
+    % check if has_x0 is true. If true inequalities are set to equalities
+    % NOTE: This adds an additional call to make_consistent, but is
+    % necessary for the check for has_x0
```
Comment (FreyJo) [inline interfaces/acados_matlab_octave/setup_AcadosOcp_from_legacy_ocp_description.m:383]: Maybe change to
`NOTE: call make_consistent to have nx available`
